tests/regression/diffsysctl/diffsysctl_test.py

```diff
@@ -0,0 +1,220 @@
+sysctl parameters - for each parameter the proc handler function is specified
+along with the data variable and all functions using it.
+For each function the expected analysis results is specified.
+This script parses the test specification and prepares testing scenarions for
+pytest.
+"""
+
+from diffkemp.llvm_ir.build_llvm import LlvmKernelBuilder
+from diffkemp.semdiff.function_diff import functions_diff, Result
+from tests.regression.module_tools import prepare_module
+import glob
+import os
+import pytest
+import yaml
+
+
+class FunctionSpec:
+    """"
+    Specification of a function in kernel along the modules/source files where
+    it is located and the expected result of the function comparison between
+    two kernel versions.
+    """
+    def __init__(self, name, result, old_module=None, new_module=None):
+        self.name = name
+        self.old_module = old_module
+        self.new_module = new_module
+        self.result = result
+
+
+specs_path = "tests/regression/diffsysctl/test_specs"
+tasks_path = "tests/regression/diffsysctl/kernel_modules"
+
+
+def collect_task_specs():
+    """Collecting and parsing YAML files with test specifications."""
+    result = list()
+    if not os.path.isdir(tasks_path):
+        os.mkdir(tasks_path)
+    cwd = os.getcwd()
+    os.chdir(specs_path)
+    for spec_file_path in glob.glob("*.yaml"):
+        with open(spec_file_path, "r") as spec_file:
+            spec_yaml = yaml.load(spec_file)
+            if "disabled" in spec_yaml and spec_yaml["disabled"] is True:
+                continue
+            for sysctl in spec_yaml["sysctls"]:
+                proc_yaml = sysctl["proc_handler"]
+                data_yaml = sysctl["data_variable"]
+
+                spec = dict(spec_yaml)
+                spec["sysctl"] = sysctl["sysctl"]
+                spec["data"] = data_yaml["name"]
+                spec["data_functions"] = data_yaml["functions"]
+                spec["proc_handler"] = proc_yaml
+
+                spec_id = os.path.splitext(spec_file_path)[0] + "_" + \
+                    sysctl["sysctl"]
+                result.append((spec_id, spec))
+
+    os.chdir(cwd)
+    return result
+
+
+specs = collect_task_specs()
+
+
+class TaskSpec:
+    """
+    Task specification representing one testing scenario (one sysctl parameter)
+    """
+    def __init__(self, spec):
+        # Values from the YAML file
+        self.old_kernel = spec["old_kernel"]
+        self.new_kernel = spec["new_kernel"]
+        self.sysctl = spec["sysctl"]
+        self.data = spec["data"]
+        self.data_functions = list()
+        for function, result in spec["data_functions"].iteritems():
+            self.data_functions.append(FunctionSpec(function, result))
+        self.proc_handler = list()
+        for function, result in spec["proc_handler"].iteritems():
+            self.proc_handler.append(FunctionSpec(function, result))
```

Comment:
There is always a single proc_handler function, no need to use iteritems() and store it in a list.